pkg/repo/migrations/v2_3.go

```diff
@@ -32,24 +32,31 @@ var V2Migration = repo.NewMigration(
 		// we load the config to resolve the libp2p node id. Loading the config this way will also
 		// use default values, args and env vars to fill in the config, so we can be sure we are
 		// reading the correct libp2p key in case the user is overriding the default value.
-		resolvedCfg, err := config.Load(repoPath)
+		newCtx := config.New()
+		if _, err := os.Stat(filepath.Join(repoPath, config.FileName)); err == nil {
+			if err := newCtx.Load(filepath.Join(repoPath, config.FileName)); err != nil {
+				return err
+			}
+		}
+		r.EnsureRepoPathsConfigured(newCtx)
+		resolvedCfg, err := newCtx.Current()
```

Comment:
I think all looks good, except ensuring repo paths are configured. Will this create the repo paths if they don't exist, or just populate the config with the paths?